tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/EngSys/CodeownersTools/CodeownersTools.cs

```diff
@@ -251,10 +254,19 @@ private async Task<List<string>> CreateCodeownersPR(
             string description, // used for commit message, PR title, and PR description
             string branchPrefix,
             string identifier,
-            string workingBranch)
+            string workingBranch,
+            string repoOwner)
         {
             List<string> resultMessages = new();
             var branchName = "";
+            var currentUsername = (await githubService.GetGitUserDetailsAsync()).Login;
+
+            if (repoOwner != Constants.AZURE_OWNER_PATH && repoOwner != currentUsername)
```

Comment:
You hit an endpoint that returned an object like 

```
  "permissions": {
    "admin": false,
    "maintain": true,
    "push": true,
    "triage": true,
    "pull": true
  },
```

Hit that endpoint and use the results from that to see if you have `push` permissions.